pkg/cache/internal/cached_resource.go

```diff
@@ -38,6 +38,11 @@ type CachedResource struct {
 	// cacheVersion is the version of the cache at the time of last update, used in sotw.
 	cacheVersion string
 
+	// onDemandOnly indicates if this resource is only sent when explicitly requested (ODCDS support).
```

Comment:
I would remove reference to the ODCDS part. As Valerian mentioned this could apply to Listeners as well.

```suggestion
	// onDemandOnly indicates if this resource is only sent when explicitly requested: for example when performing ODCDS (on-demand cluster discovery).
```

Reply:
That's fair - Making it more use case agnostic is probably better for when we upstream this.